### PR TITLE
Pinned yarn package manager to stable version

### LINKED ISSUE #1931: Too many levels of symbolic links
**Bug description**
Cannot build nginx image based on guideline in https://github.com/DefectDojo/django-DefectDojo#quick-start and always showing error `Too many levels of symbolic links`

**Steps to reproduce**

1. - git clone https://github.com/DefectDojo/django-DefectDojo
2. - cd django-DefectDojo
3. - docker-compose build

**Expected behavior**
Build success.

**Deployment method** *(select with an `X`)*
- [ ] Kubernetes
- [X] Docker
- [ ] setup.bash / legacy-setup.bash

**Environment information**
 - Operating System: CentOS Linux release 7.7.1908 (Core) and also happen in macOS Catalina 10.15.3
 - DefectDojo Commit Message: d411791: Merge pull request #1834 from ptrovatelli/v1.5.4rc6 [2020-02-24 16:13:31 -0600]

**Console logs** (optional)
```
 ---> Running in b9f7d039d07a
yarn install v1.22.0
warning package.json: No license field
info No lockfile found.
warning No license field
[1/5] Validating package.json...
warning No license field
[2/5] Resolving packages...
warning @yarn_components/mocha > jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
warning @yarn_components/mocha > glob > minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
warning @yarn_components/mocha > glob > graceful-fs@2.0.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
[3/5] Fetching packages...
[4/5] Linking dependencies...
warning " > @yarn_components/metisMenu@3.0.5" has unmet peer dependency "jquery@^3.3.1".
[5/5] Building fresh packages...
success Saved lockfile.
Done in 73.17s.
yarn cache v1.22.0
warning package.json: No license field
success Cleared cache.
Done in 0.93s.
Traceback (most recent call last):
  File "manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.5/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 188, in handle
    collected = self.collect()
  File "/usr/local/lib/python3.5/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 105, in collect
    for path, storage in finder.list(self.ignore_patterns):
  File "/usr/local/lib/python3.5/site-packages/django/contrib/staticfiles/finders.py", line 131, in list
    for path in utils.get_files(storage, ignore_patterns):
  File "/usr/local/lib/python3.5/site-packages/django/contrib/staticfiles/utils.py", line 39, in get_files
    yield from get_files(storage, ignore_patterns, dir)
  File "/usr/local/lib/python3.5/site-packages/django/contrib/staticfiles/utils.py", line 39, in get_files
    yield from get_files(storage, ignore_patterns, dir)
  File "/usr/local/lib/python3.5/site-packages/django/contrib/staticfiles/utils.py", line 23, in get_files
    directories, files = storage.listdir(location)
  File "/usr/local/lib/python3.5/site-packages/django/core/files/storage.py", line 316, in listdir
    if entry.is_dir():
OSError: [Errno 40] Too many levels of symbolic links: '/app/components/node_modules/@yarn_components/moment/meteor/moment.js'
ERROR: Service 'nginx' failed to build: The command '/bin/sh -c cd components &&   yarn && yarn cache clean &&   cd .. &&   python3 manage.py collectstatic --noinput &&   true' returned a non-zero code: 1
```

### PR DESCRIPTION
**Pull request by arkwrn**
- Updating nodejs from 11.x to 12.x due to decprecation warning, since nodejs 11.x is no longer actively supported
- Freeze yarn package manager to version 1.21.0 (stable version) due to issue when building nginx image

Fixes #1931

### CODE REVIEW

**Comment by valentijnscholten** [comment]
@arkwrn could you resolve the merge conflict?

**Comment by arkwrn** [comment]
> @arkwrn could you resolve the merge conflict?

Done @valentijnscholten

**Comment by valentijnscholten** [comment]
I think pinning is good, but shouldn't we pin it in all loations then? Also the other install scripts, `package.json, etc?

**Comment by arkwrn** [comment]
Agree, I'll check it tomorrow and get back to you as soon as i figure it out.

File: Dockerfile.nginx [hunk 1]
```diff
@@ -30,16 +30,17 @@ ENV \
...
+  wget https://github.com/yarnpkg/yarn/releases/download/v1.21.0/yarn_1.21.0_all.deb && \
+  dpkg -i yarn_1.21.0_all.deb && \
   apt-get -y update && \
+  echo "$(yarn --version)" && \
   apt-get -y install nodejs && \
   echo "$(node --version)" && \
-  apt-get -y install --no-install-recommends yarn && \
-  echo "$(yarn --version)" && \
   apt-get clean && \
+  rm yarn_1.21.0_all.deb && \
```
Comment (TheDKSU) [inline Dockerfile.nginx]: This fixes #1931

**Comment by valentijnscholten** [comment]
I think I'd prefer to have some yarn version that gets updated automatically with bugfixes/security fixes i.e. minor updates i.e. the version that comes with debian from their official apt repo.
Don't know if that is possible, but all in all I think this PR makes it better by at least having something more stable.

**Comment by ptrovatelli** [comment]
@valentijnscholten i'm all for pinning; dependabot is there to upgrade the version. we should aim for stability first to avoid issues after releasing like we had recently.

**Pull request merged**